objective-c/README.md

```diff
@@ -0,0 +1,522 @@
+# Objective-c 
+
+## Introduction
+
+This document is useful if you work on a huge project and want to keep it readable, clean & maintainable.
+Also helps you to avoid common mistakes (I have found since I started my career) and shows some best-practices.
+
+## Class naming convention
+
+Use class prefixes instead of just plain names. This makes your classes more unique - imagine if everybody would use plain names: you couldn't import a third party framework without conflict.
+> You can set up the default prefix in Xcode: select the project file in the Navigator (file browser, left side),
+> select File inspector tab and under Project Document / Class Prefix.
+
+## Markers (pragma mark)
+
+These markers can help your (and an other developer's) work to find methods.
+
+Recommended markers in order:
+
+* Initialization (-init, -dealloc, shared instance accessor of a singleton)
+
+* Private methods
+
+* View lifecycle (overrided methods of UIKit classes: -viewDidLoad, -viewWillAppear... etc.)
+
+* Public methods
+
+* Overrides
+
+* Accessors
+
+* Notification handlers
+
+* KVO handler
+
+* <[DelegateName]> (for delegate method implementations)
+
+Usage:
+`#pragma mark - [title]`
+
+or embed markers:
+
+`#pragma mark - [main title]`
+
+`#pragma mark [sub-title]`
+
+## Naming global constants
+
+Just like notification names, enum types... etc. They need to reflect to the container class, so 
+it should use it as a base. This help the user of this class to use code completion without checking the classes header.
+
+Example:
+
+	extern NSString * const MYClassDidSomethingNotification;
+
+	extern const CGFloat MYClassConstantFloatValue;
+
+	typedef NS_ENUM(NSInteger, MYClassSide)
+	{
+	MYClassSideLeft,
+	MYClassSidesRight
+	};
+
+	@interface MyClass : ...
+
+
+## Asserts
+
+Use asserts to keep thigt conditions in DEBUG mode. It will help you to find programming mistakes and/or
+invalid states in runtime.
+
+Examples:
+
+	- (BOOL)checkThis:(NSNumber *)number
+	{
+		NSParameterAssert(number);
+		...
+	}
+
+	- (void)myMethod
+	{
+		const NSUInteger numberOfItems = [self numberOfItems];
+		
+		NSAssert(0 < numberOfItems, @"No item has been found");
+		
+		...
+	}
+
+Handy, isn't it?
+
+> Avoid to use messages in the NSAssert body!
```

Comment:
Also don't depend on `NSAssert` to catch things at runtime in Release mode.  They're completely compiled out so if you hit that situation in a Release mode build, your program will continue on in a state you were trying to prevent.